x-pack/plugins/spaces/server/routes/api/external/update_objects_spaces.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import intersection from 'lodash/intersection';
+
+import { schema } from '@kbn/config-schema';
+
+import { ALL_SPACES_ID } from '../../../../common/constants';
+import { wrapError } from '../../../lib/errors';
+import { SPACE_ID_REGEX } from '../../../lib/space_schema';
+import { createLicensedRouteHandler } from '../../lib';
+import type { ExternalRouteDeps } from './';
+
+export function initUpdateObjectsSpacesApi(deps: ExternalRouteDeps) {
+  const { externalRouter, getStartServices } = deps;
+
+  const spacesSchema = schema.arrayOf(
+    schema.string({
+      validate: (value) => {
+        if (value !== ALL_SPACES_ID && !SPACE_ID_REGEX.test(value)) {
+          return `lower case, a-z, 0-9, "_", and "-" are allowed, OR "*"`;
+        }
+      },
+    }),
+    {
+      validate: (spaceIds) => {
+        if (uniq(spaceIds).length !== spaceIds.length) {
+          return 'duplicate space ids are not allowed';
+        }
+      },
+    }
+  );
+
+  externalRouter.post(
+    {
+      path: '/api/spaces/_update_objects_spaces',
+      validate: {
+        body: schema.object(
+          {
+            objects: schema.arrayOf(schema.object({ type: schema.string(), id: schema.string() })),
+            spacesToAdd: spacesSchema,
+            spacesToRemove: spacesSchema,
+          },
+          {
+            validate: ({ spacesToAdd, spacesToRemove }) => {
+              if (!spacesToAdd.length && !spacesToRemove.length) {
+                return 'spacesToAdd and/or spacesToRemove must be a non-empty array of strings';
+              }
+              if (intersection(spacesToAdd, spacesToRemove).length > 0) {
+                return 'spacesToAdd and spacesToRemove cannot contain any of the same strings';
+              }
```

Comment:
NIT: aren't we asserting that already in the underlying implementation?

Reply:
Yeah that's true, I can remove this validation.